open-craft/learner-profile-dashboard
# Address post-delivery feedback, increase logging output

cf. [OC-4615](https://tasks.opencraft.com/browse/OC-4615), [OC-4751](https://tasks.opencraft.com/browse/OC-4751)

This PR extends the amount of logging that the LPD does around sending learner profile data to the adaptive engine while improving existing logging behavior (168b6ed7d03d3314bd949bcd63d1be0d906f6dea), and addresses additional feedback from the TLL team (8dcb8582a15ad8f390b28b4aac9a65c2f85e0be5). It also updates some terminology to better differentiate between the concepts of (answer) values and scores (8b1bb86351420582827b93ef7ba7cd39bce6c1d3). Lastly, it brings the documentation up-to-date with additions that we made since #12.

**Related PRs** (increase logging output)

- https://github.com/open-craft/ansible-playbooks/pull/51
- https://github.com/open-craft/ansible-secrets/pull/58

**Test instructions**

1. Update the framing text of a question to include one or more HTML entities.

1. Access the LPD and observe that the HTML entities render correctly.

1. Tail `debug.log` and submit different combinations of answers. Observe that the LPD now:
    - Logs information about learner profile data to transmit to the adaptive engine.
    - Pretty-prints data structures such as lists and dictionaries. 
    - Includes error tracebacks in logs (instead of just briefly mentioning a given error).
    - Explicitly mentions date and time of latest submission (if successful).
    - Logs information about scores to create or update.

**Reviewers**
